.github/workflows/build.yml

```diff
@@ -500,6 +500,7 @@ jobs:
       options: ./configure --config-cache --disable-gil --with-thread-sanitizer --with-pydebug
       suppressions_path: Tools/tsan/suppressions_free_threading.txt
       tsan_logs_artifact_name: tsan-logs-free-threading
+      os-matrix: '["ubuntu-22.04", "macos-14"]'
```

Comment:
To avoid queuing up too many jobs for the limited number of macOS builders, we'd want to use `macos-14` for forks, but Cirrus for upstream builds.

See elsewhere in this file for examples, also needs changes in the reusable workflow.